usefulUtilities/botPlayer/assignmentClientPlayer.js

```diff
@@ -0,0 +1,264 @@
+            if (success) {
+                _this.isPlayingRecording = true;
+
+                Users.disableIgnoreRadius();
+
+                Agent.isAvatar = true;
+
+                Recording.setPlayFromCurrentLocation(false);
+                Recording.setPlayerUseDisplayName(true);
+                Recording.setPlayerUseHeadModel(false);
+                Recording.setPlayerUseAttachments(true);
+                Recording.setPlayerLoop(false);
+                Recording.setPlayerUseSkeletonModel(true);
+                Recording.setPlayerTime(0.0);
+                Recording.setPlayerVolume(1.0);
+                
+                Recording.startPlaying();
+            } else {
+                console.log("Could not load recording " + fileToPlay);
+
+                _this.isPlayingRecording = false;
+                _this.recordingFilename = "";
+                // This should remove the avatars however they are coming back in as white spheres at the origin
+                // Agent.isAvatar = false;
+            }
+        });
+    }
+
+
+    // Stop the bot and remove 
+    // #NOTE:  Adding and removing the agent is currently causing issues.  Using a work around for the meantime
+    function stop() {
+        console.log("Stop playing " + this.recordingFilename);
+
+        if (Recording.isPlaying()) {
+            Recording.stopPlaying();
+
+            // This looks like it's a platform bug that this can't be removed
+            // Agent.isAvatar = false;
+        }
+        this.isPlayingRecording = false;
+    }
+
+    
+    // Check if the bot is playing
+    function isPlaying() {
+        return this.isPlayingRecording;
+    }
+
+
+    Player.prototype = {
+        play: play,
+        stop: stop,
+        isPlaying: isPlaying
+    };
+
+    
+    // #endregion
+    // *************************************
+    // END PLAYER
+    // *************************************
+
+    // *************************************
+    // START MESSAGES
+    // *************************************
+    // #region MESSAGES
+    
+
+    // Handle messages fromt he manager
+    var IGNORE_PLAYER_MESSAGES = ["REGISTER_ME", "ARE_YOU_THERE_MANAGER"];
+    var playInterval = null;
+    var INTERVAL_TIME = 20000;
+    function onMessageReceived(channel, message, sender) {
+        if (channel !== ASSIGNMENT_MANAGER_CHANNEL || 
+            sender === scriptUUID || 
+            IGNORE_PLAYER_MESSAGES.indexOf(message.action) > -1) {
+            return;
+        }
+
+        try {
+            message = JSON.parse(message);
+        } catch (e) {
+            console.log("Can not parse message object");
+            console.log(e);
+
+            return;
+        }
+
+        if (message.uuid !== scriptUUID) {
+            return;
+        }
+
+        switch (message.action){
+            case "PLAY":
+                if (!player.isPlaying()) {
+                    var randoInterval = Math.max(Math.floor(Math.random() * INTERVAL_TIME), 7500);
```

Comment:
This section is what we were playing with.  Probably could be finessed a bit better.